database/duckdb/duckdb.go

```diff
@@ -0,0 +1,204 @@
+package duckdb
+
+import (
+	"database/sql"
+	"fmt"
+	"io"
+	nurl "net/url"
+	"strings"
+
+	"go.uber.org/atomic"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database"
+	"github.com/hashicorp/go-multierror"
+	_ "github.com/marcboeker/go-duckdb"
+)
+
+func init() {
+	database.Register("duckdb", &DuckDB{})
+}
+
+const MigrationTable = "gmg_schema_migrations"
+
+type DuckDB struct {
+	db       *sql.DB
+	isLocked atomic.Bool
+}
+
+func (d *DuckDB) Open(url string) (database.Driver, error) {
+	purl, err := nurl.Parse(url)
+	if err != nil {
+		return nil, fmt.Errorf("parsing url: %w", err)
+	}
+	dbfile := strings.Replace(migrate.FilterCustomQuery(purl).String(), "duckdb://", "", 1)
+	db, err := sql.Open("duckdb", dbfile)
+	if err != nil {
+		return nil, fmt.Errorf("opening '%s': %w", dbfile, err)
+	}
+
+	if err := db.Ping(); err != nil {
+		return nil, fmt.Errorf("pinging: %w", err)
+	}
+	d.db = db
+
+	if err := d.ensureVersionTable(); err != nil {
+		return nil, fmt.Errorf("ensuring version table: %w", err)
+	}
+
+	return d, nil
+}
+
+func (d *DuckDB) Close() error {
+	return d.db.Close()
+}
+
+func (d *DuckDB) Lock() error {
+	if !d.isLocked.CAS(false, true) {
```

Comment:
【NITS】
CAS looks deprecated.
https://github.com/uber-go/atomic/blob/df7bc75a1bb43fdc1f905256ac5b387b332b0028/bool.go#L59

(I'm not a maintainer. I point this because it should save maintainers' time. Thank you for this PR and I'll be happy if it will be merged!)

Reply:
Thank you! I'll see if I can get a team member assigned on my side to get this updated.